Runtimes/Supplemental/Differentiation/CMakeLists.txt

```diff
@@ -0,0 +1,72 @@
+cmake_minimum_required(VERSION 3.29)
+
+set(CMAKE_POSITION_INDEPENDENT_CODE YES)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/../cmake/modules")
+
+if($ENV{BUILD_NUMBER})
+  math(EXPR BUILD_NUMBER "$ENV{BUILD_NUMBER} % 65535")
+  set(BUILD_NUMBER ".${BUILD_NUMBER}")
+endif()
+project(SwiftDifferentiation
+  LANGUAGES Swift C
+  VERSION 6.1.0${BUILD_NUMBER})
+
+if(NOT PROJECT_IS_TOP_LEVEL)
+  message(SEND_ERROR "Swift Differentiation must build as a standalone project")
+endif()
+
+set(${PROJECT_NAME}_SWIFTC_SOURCE_DIR
+  "${PROJECT_SOURCE_DIR}/../../../"
+  CACHE FILEPATH "Path to the root source directory of the Swift compiler")
+
+find_package(SwiftCore)
+find_package(math)
+
+include(gyb)
+
+option(SwiftDifferentiation_ENABLE_VECTOR_TYPES "Enable vector support"
+  ${SwiftCore_ENABLE_VECTOR_TYPES})
+
+option(SwiftDifferentiation_ENABLE_LIBRARY_EVOLUTION "Enable library evolution"
+  ${SwiftCore_ENABLE_LIBRARY_EVOLUTION})
+
+add_compile_options(
+  $<$<COMPILE_LANGUAGE:Swift>:-explicit-module-build>
+  $<$<COMPILE_LANGUAGE:Swift>:-nostdlibimport>
+  $<$<COMPILE_LANGUAGE:Swift>:-parse-stdlib>
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-library-level api>"
+  $<$<COMPILE_LANGUAGE:Swift>:-enforce-exclusivity=unchecked>
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-target-min-inlining-version min>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature NoncopyableGenerics2>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature SuppressedAssociatedTypes>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature SE427NoInferenceOnExtension>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature NonescapableTypes>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature LifetimeDependence>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature MemberImportVisibility>"
+  $<$<AND:$<BOOL:${SwiftDifferentiation_ENABLE_LIBRARY_EVOLUTION}>,$<COMPILE_LANGUAGE:Swift>>:-enable-library-evolution>)
+
+if(SwiftDifferentiation_ENABLE_VECTOR_TYPES)
```

Comment:
We should probably put the sources under a `Sources` directory. While that is not required for CMake to work properly, I think that it would be nice to have the layout be similar to other projects.

Reply:
I'm not sure if I understand correctly your point -- are you suggesting to have a directory structure in which the sources are not mingled with the CMake code, e.g.

* Supplemental
    * Differentiation
        * CMakeLists.txt
        * Sources
            * AnyDifferentiable.swift
            * ...
    * Synchronization
        * CMakeLists.txt
        * Sources
    * ...
    * cmake/modules

Reply:
You would still have the intermingled CMakeLists.txt, but the idea is that we could have the same structure across all the libraries. I think that @etcwilde suggested that we do that subsequently, which I am fine with.